### ISSUE TITLE
Error when opening Metabolights study MTBLS719

### ISSUE THREAD

**Issue by pkrog** [ISA-tools/isa-api#446]
When trying to open MTBLS719, I get the following error:
```
Traceback (most recent call last):
  File "/home/pr228844/dev/isa2w4m/tests/../isa2w4m.py", line 8, in <module>
    isatab2w4m.main()
  File "/home/pr228844/dev/isa2w4m/test-venv/lib/python3.8/site-packages/isatools/convert/isatab2w4m.py", line 590, in main
    convert(input_dir=args_dict['input_dir'],
  File "/home/pr228844/dev/isa2w4m/test-venv/lib/python3.8/site-packages/isatools/convert/isatab2w4m.py", line 569, in convert
    assays = convert2w4m(input_dir=input_dir,
  File "/home/pr228844/dev/isa2w4m/test-venv/lib/python3.8/site-packages/isatools/convert/isatab2w4m.py", line 454, in convert2w4m
    study = select_study(investigation_file, study_filename)
  File "/home/pr228844/dev/isa2w4m/test-venv/lib/python3.8/site-packages/isatools/convert/isatab2w4m.py", line 139, in select_study
    investigation = load_investigation(investigation_file)
  File "/home/pr228844/dev/isa2w4m/test-venv/lib/python3.8/site-packages/isatools/convert/isatab2w4m.py", line 351, in load_investigation
    investigation = ISATAB.load(f)
  File "/home/pr228844/dev/isa2w4m/test-venv/lib/python3.8/site-packages/isatools/isatab.py", line 4902, in load
    ProcessSequenceFactory(
  File "/home/pr228844/dev/isa2w4m/test-venv/lib/python3.8/site-packages/isatools/isatab.py", line 5770, in create_from_df
    raise ValueError(
ValueError: ('Could not resolve Protocol parameter from Parameter Value ', 'Extraction Method')
```
Is it an error from `isatools` or from the study?

**Comment by pkrog** [comment]
@yguitton told me that the error may be inside the Metabolights study. I will wait for his feedback to close this issue.

**Comment by pkrog** [comment]
Hi @proccaserra , thanks, The study MTBLS719 has already been corrected according to @yguitton, so yes you can check with isatools if you wish on your side, but we will run checks again ourselves too.

**Comment by proccaserra** [comment]
I confirm the "Extraction Method" parameter is missing in the i_investigation.txt file served by Metabolights.
I am in touch with them regarding other issues so I'll let them know:

![image](https://user-images.githubusercontent.com/517736/165576572-1bff350b-0959-4ae4-ab6b-85f86aa379c7.png)

I'll check the parsing on my end too 

thx for the heads up

**Comment by pkrog** [comment]
With the current `MTBLS719.zip` file from Metabolights, I get the following error with `isa2w4m`:
```
2022-04-27 18:47:00,024 [INFO]: isatab2w4m.py(info:63) >> INFO: Found data file "m_MTBLS719_LC-MS_positive_reverse-phase_metabolite_profiling_v2_maf.tsv".
Traceback (most recent call last):
  File "/home/pierrick/dev/isa2w4m/tests/../isa2w4m.py", line 8, in <module>
    isatab2w4m.main()
  File "/home/pierrick/dev/isa2w4m/test39-venv/lib/python3.9/site-packages/isatools/convert/isatab2w4m.py", line 590, in main
    convert(input_dir=args_dict['input_dir'],
  File "/home/pierrick/dev/isa2w4m/test39-venv/lib/python3.9/site-packages/isatools/convert/isatab2w4m.py", line 569, in convert
    assays = convert2w4m(input_dir=input_dir,
  File "/home/pierrick/dev/isa2w4m/test39-venv/lib/python3.9/site-packages/isatools/convert/isatab2w4m.py", line 474, in convert2w4m
    sample_metadata = make_sample_metadata(study_df=study_df,
  File "/home/pierrick/dev/isa2w4m/test39-venv/lib/python3.9/site-packages/isatools/convert/isatab2w4m.py", line 391, in make_sample_metadata
    norm_sample_names = make_names(sample_names, uniq=True)
  File "/home/pierrick/dev/isa2w4m/test39-venv/lib/python3.9/site-packages/isatools/convert/isatab2w4m.py", line 258, in make_names
    v = u[:]
TypeError: 'NoneType' object is not subscriptable
make: *** [Makefile:53: test39] Error 1
```
So I guess something is missing to build the Sample Metadata table.

**Comment by proccaserra** [comment]
@pkrog investigating further with MTBLS719, there seems to be an issue with the NMR Metabolights assay definition, which is used by the web application to create the ISA document. As far as I can tell the problems are as following:
- protocol parameter used but not declared or attached to the wrong protocol. This protocol parameter is 'Extract Method'
- incorrect ISA field order (these tables aren't create with the ISA-API), for instance "Protocol REF" field found *between* 'Data Transformation Name' and 'Derived Data File'
- ISA Node fields (e.g. Extract Name, NMR Assay Name, Data Transformation Name) with no values. This leads to parsing error or failure with the python ISA-API and to validation failure reported by the java ISA validator. 

Currently, document all the problems and will be reporting the Metabolights.

**Comment by yguitton** [comment]
Thanks for your help @proccaserra and @pkrog for the time and effort to solve this.

**Comment by proccaserra** [comment]
@yguitton @pkrog I have sent Metabolights amended ISA-Tab files. I'll send those to you both now.
I didn't have time yet to look at the isa2w4m.py and MAF file issue. bear with me!

**Comment by proccaserra** [comment]
@yguitton @pkrog update:
There is another issue with this submission (you probably ran into it)

This submission where the current of the python ISA-validator sends a warning message "Duplicate Characteristics, dropping Fields from file...".

After some digging, I realised that while the code was correctly detecting an issue, it did not report in a particularly helpful way. 
We have now fixed that in a branch (not yet merged in our main) but the report will now include the offending affected samples and the problematic annotations.

Essentially, the issue boils down to one same sample bearing 2 distinct values for the same property (e.g. 'A' and 'B' for characteristics X at the same time, when a specific unique value is expected).

In MTBLS719, the issue presents itself for Source `MS_U_RNEG_SR` which is annotated simultaneously as 'Study Reference' and "Long Term Reference".

![image](https://user-images.githubusercontent.com/517736/184078084-e903e05c-ddd8-4bad-93f0-58740ea2312f.png)
